src/config.rs

```diff
@@ -312,3 +314,57 @@ impl DerefMut for IdentFormats {
 pub enum IdentFormatsTheme {
     Legacy,
 }
+
+#[cfg(feature = "unstable-riscv")]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize), serde(default))]
+#[derive(Clone, PartialEq, Eq, Debug, Default)]
+#[non_exhaustive]
+pub struct RiscvConfig {
+    pub core_interrupts: Option<Vec<RiscvEnumItem>>,
+    pub exceptions: Option<Vec<RiscvEnumItem>>,
+    pub priorities: Option<Vec<RiscvEnumItem>>,
+    pub harts: Option<Vec<RiscvEnumItem>>,
```

Comment:
Is `None` different from `Some(Vec::empty)` here?

Reply:
Currently, if you define `core_interrupts` in your YAML file, `svd2rust` will generate a custom `CoreInterrupt` struct for your target, regardless of whether it is empty or not. Thus, an empty vector would probably be meaningless. I can change this behavior following any of your design decisions.

Reply:
I want to say that you can leave just `Vec<T>` and check `vec.is_empty()` instead of `option.is_some`.

Possibly you should also mark a field with `serde(default)`. Not sure.

Reply:
Will take a look to your suggestion!